src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -77,6 +81,33 @@ export default function getAdaptationSwitchStrategy(
     return { type: "continue", value: undefined };
   }
 
+  if (adaptation.type === "audio" && audioTrackSwitchingMode === "direct") {
+    const playingBufferedChunk = arrayFind(inventory, (buf) =>
+      playbackInfo.currentTime >= buf.start && playbackInfo.currentTime <= buf.end);
+    if (playingBufferedChunk === undefined) {
```

Comment:
I begin to think that there is a little risk here of not reloading, if `currentTime` is not exactly aligned with the `SegmentInventory`.

Also, we don't check that we're currently playing the current Period, so we could be reloading even if that's the next Period (that is being pre-loaded for now) that needs to be reloaded.

I think that for the code for the video (below) can be _mostly_ re-used.
```ts
      // We're playing the current Period
      isTimeInRange({ start, end }, currentTime) &&
      // There is data for the current position
      playbackInfo.readyState > 1 &&
      // We're not playing the current wanted video Adaptation
      !isTimeInRanges(adaptationInBuffer, currentTime))
```
Here:
  1. we ensure the current Period is being played
  2. we ensure that we are playing content - to avoid an unnecessary reload if we're not
      __We could remove that one now that I think of it, as we could have problem when there are incompatible codecs__, I think I'll might even remove it for video before the next release and replace it by the same codec check we will do here
  3. we ensure that the new AdaptationSet is not already being buffered

Reply:
If I understand correctly, with your evolve on the Segment Buffer (codec properties), I no longer need to check the `currentTime` right ? since the Segment buffer provides us the necessary?

Reply:
You don't need to check `currentTime` to know the codec of what is playing but you might still want to check the `currentTime` to see if we're currently playing the corresponding Period.